quinn-proto/src/connection.rs

```diff
@@ -1203,13 +1201,14 @@ impl Connection {
                     ctx.incoming_handshakes -= 1;
                 }
                 let n = self.get_tx_number();
+                debug_assert!(n < 64); // should fit in one byte
```

Comment:
`handshake_close()` doesn't have access to `largest_acked_packet`, so it cannot use `PacketNumber::new()`. Since `handshake_close()` is all about the handshake, it seemed reasonable that it can simply default to the single-byte encoding of packet numbers instead. I'll clarify the comment here.

Reply:
> handshake_close() doesn't have access to largest_acked_packet, so it cannot use PacketNumber::new().

Perhaps it should take a `PacketNumber`?

> Since handshake_close() is all about the handshake, it seemed reasonable that it can simply default to the single-byte encoding of packet numbers instead.

I'm not sure this is reliable. Handshakes can potentially be large due to e.g. large certificate chains, server transport parameters, or future extensions.

Reply:
> Perhaps it should take a `PacketNumber`?

That's what I started to do, but then it also needs the raw packet number (as it also doesn't have the state to expand the packet number again) to be able to encrypt the payload.

As I understand it, a normal handshake completes in substantially less than packets per peer, so it seems to me that 6x that ought to provide a safe margin.

Reply:
Is there a firm guarantee anywhere? I'd rather not generate invalid close packets in weird-but-legal situations.